### PR TITLE
Refactor deletion test

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

Refactor deletion test to not need multiple attempts, instead collecting results and returning an aggregation.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing

## Test manual

 - CI

### CODE REVIEW

File: fleetshard/pkg/centralreconciler/reconciler.go [hunk 1]
```diff
@@ -147,17 +147,27 @@ func (r *CentralReconciler) Reconcile(ctx context.Context, remoteCentral private
...
 	}
-	if crDeleted, err := r.ensureCentralCRDeleted(ctx, central); err != nil || !crDeleted {
+	globalDeleted = routeDeleted && globalDeleted
+
+	centralDeleted, err := r.ensureCentralCRDeleted(ctx, central)
+	if err != nil {
 		return false, err
 	}
-	if namespaceDeleted, err := r.ensureNamespaceDeleted(ctx, central.GetNamespace()); err != nil || !namespaceDeleted {
+	globalDeleted = routeDeleted && centralDeleted
```
Comment (kovayur) [inline fleetshard/pkg/centralreconciler/reconciler.go:161]: Did you mean this?
```suggestion
	globalDeleted = globalDeleted && centralDeleted
```

Comment (SimonBaeumer) [reply fleetshard/pkg/centralreconciler/reconciler.go:162]: yes, done

File: fleetshard/pkg/centralreconciler/reconciler.go [hunk 2]
```diff
@@ -147,17 +147,27 @@ func (r *CentralReconciler) Reconcile(ctx context.Context, remoteCentral private
...
+
+	nsDeleted, err := r.ensureNamespaceDeleted(ctx, central.GetNamespace())
+	if err != nil {
 		return false, err
 	}
+	globalDeleted = routeDeleted && nsDeleted
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply fleetshard/pkg/centralreconciler/reconciler.go:168]: done

**Review by kovayur — APPROVED** [review]
LGTM

**Pull request merged**